hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -467,6 +467,9 @@ public void refreshSources(String peerId) throws IOException {
       ReplicationSourceInterface toRemove = this.sources.put(peerId, src);
       if (toRemove != null) {
         LOG.info("Terminate replication source for " + toRemove.getPeerId());
+        // Reset sizeOfLogQueue, log will re enqueue to the created new source.
+        toRemove.getSourceMetrics()
```

Comment:
@frostruan  Do you think we need to reset just size of log queue metric or other source metrics also? IMO we should reset all the metrics and start with clean state?

Reply:
Thanks for reviewing @shahrs87 

Currently we monitor the replication status by the sizeOfLogQueue metric, so I just reset the sizeOfLogQueue here. I'll check if there is any other metrics need to be reset. Thanks for your suggestion.

Reply:
Are there any rules about which metrics should be reset while others should not?

Reply:
Thanks for review Duo.@Apache9

Sorry, currently have no idea about the rule you mentioned, maybe the rule is a little complicated.

I think what makes the rule complicated is that we first create the new ReplicationSource, then replace the old ReplicationSource with the new one, and then if the old ReplicationSource exists,  terminate it. And since HBASE-23231, we will not clear the old metrics.

If we adjust the order, first terminate the old ReplicationSource if exists and then create and register the new ReplicationSource, the logic of metric here maybe will be simpler, and of course we can clear the metric when terminate the old ReplicationSource. But I'm not sure yet,  still need to confirm

Reply:
This change will effectively undo [HBASE-23231](https://issues.apache.org/jira/browse/HBASE-23231) @Apache9  Since you were one of the reviewer on HBASE-23231, do you think it is safe to do this?